client/lcd/docs/docs.go

```diff
@@ -0,0 +1,725 @@
+// GENERATED BY THE COMMAND ABOVE; DO NOT EDIT
```

Comment:
Is this from a previous consensus on API documentation - can you link to it?

Reply:
Yes!

Reply:
This is something that has been discussed repeatedly in a number of issues. Having automatically generated API docs is going to be increasingly important as the surface area of the API grows. I've also personally used swagger in previous projects and found it amazing.

Reply:
Generally agreed - but we should figure out how exactly we want to do this. If we can now generate these automatically:

1. Are the downstream consumers (@faboweb @fedekunze) in favor of this particular autogeneration format / will it suit their needs?
2. Why does this PR commit an autogenerated file?
3. Can we remove the existing manual docs in favor of autogenerated ones?

Reply:
1. Yes, see issues
2. It shouldn't, we should add that to `.gitignore`
3. Yes, eventually.

Reply:
Great, thanks for the clarification. Sounds like @faboweb or @fedekunze would be far better suited than I to review this PR!

Reply:
@cwgoes @jackzampolin @fedekunze 
Please refer to this [link](https://github.com/swaggo/swag) for swag tool. Currenly it has some defects. It can't properly generate the data structure for some complex self-define struct. Besides, I can find all APIs when theses API don't locate in a same root folds. So I have to manually edit docs.go

Reply:
I don't think we want to be in a position where we are relying on a non working and manual fixing for our docs process. Can't we just use a `swagger.yaml` file and update that instead of relying on go struct tags and the `swag` tool?

Reply:
Please refer this [swag](https://github.com/swaggo/swag/tree/master/example/basic) example. It requres the main.go locate in the root fold. And all related handlers and data struct also locate in this root folds. However, in cosmos-sdk, main.go locates in cmd/gaia/cmd/gaiacli, and handlers locate in many folds, such as client/key, x/. And data strcut locates in many folds too. 
So if we want to make swag generation works well, we have to refactor our code structure. Do you agree with me @fedekunze ? 
I'm thinking usging swagger.yaml file. But it may also require maunally edit API docs.
In long run, maybe refactor our code strcuture is a good idea.